### ISSUE TITLE
[NEXT-902] unable to disable fetch-cache or revalidate

### ISSUE THREAD

**Issue by jwalcher** [vercel/next.js#47273]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.16.0
      npm: 8.11.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.2.5-canary.8
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Data fetching (gS(S)P, getInitialProps)

### Link to the code that reproduces this issue

https://github.com/jwalcher/no-no-store/

### To Reproduce

- execute `yarn run build`, `yarn run start`, navigate to `localhost:3000` 
- navigate to `localhost:3000/api/revalidate` and back to `localhost:3000`, check whether content updates
- uncomment various cache config options in `page.tsx`, rebuild and refresh/revalidate 
- empty `.next/cache/fetch-cache` as appropriate and retry

### Describe the Bug

- `response.revalidate` API does not work in latest 13.2.5 canary, but did up to and including 13.1.6
- `export const dynamic = “force-dynamic”`: data still cached indefinitely
- `export const fetchCache = "force-no-store"`: data still cached indefinitely
- `export const revalidate = 10`: data revalidated after ~10 secs
- `fetch(..., { cache: "no-store" })`: data not cached

### Expected Behavior

all methods above should lead to data revalidation as per latest beta docs:
- caching: https://beta.nextjs.org/docs/data-fetching/caching
- per segment configuration: https://beta.nextjs.org/docs/api-reference/segment-config#dynamic
- on-demand revalidation: https://beta.nextjs.org/docs/data-fetching/revalidating

<sub>[NEXT-902](https://linear.app/vercel/issue/NEXT-902/unable-to-disable-fetch-cache-or-revalidate)</sub>

**Comment by medonat** [comment]
Hi, is it possible to share this workaround here. We have the same problem and it would be great if we could get it working.

**Comment by jwalcher** [comment]
the main idea is in `pages/api/revalidate.jsx`
`import fs from "fs";
import path from "path";
const directory = process.cwd() + "/.next/cache/fetch-cache";
export default async function handler(req, res) {
    console.log("clearing cache");
    fs.readdir(directory, (err, files) => {
      if (err) throw err;
      for (const file of files) {
           fs.unlink(path.join(directory, file), (err) => {
          if (err) throw err;
        });}});
    if (process.env.NODE_ENV == "production") {
      console.log("revalidating");
      await res.revalidate("/");
      return res.json({ revalidated: true });
    } else {
      return res.json({ cacheCleared: true });
    }
}`
but the whole thing **only** work if we also say
` experimental: {
    appDir: true,
    isrMemoryCacheSize: 50
  }`
in `next.confg.js`. My sample page is
`app/page.tsx` 
`export default async function Home() {
  const res = await fetch("https://www.timeapi.io/api/Time/current/zone?timeZone=Europe/Amsterdam");
  const data = (await res.json());
  return (
      <h1>
        Time in Amsterdam is {data.hour}:{data.minute}:{data.seconds}
      </h1>
)}`
But i also tested this successfully with a GraphQL endpoint/Apollo client

**Comment by t3dotgg** [comment]
Can confirm I just lost most of my Saturday to this issue

**Comment by Gyurmatag** [comment]
Is this working now? Can someone confirm?

**Comment by karlhorky** [comment]
@jwalcher confirmed over here: https://github.com/vercel/next.js/discussions/42290#discussioncomment-5502200

**Comment by theonlydaleking** [comment]
> Can confirm I just lost most of my Saturday to this issue

Me too (Well Friday here) 

Can confirm this is now fixed with 13.3.0,

**Comment by mishabak** [comment]
Hello,
   I'm using  eg:- fetch('https://...', { next: { revalidate: 60 } });
  update  in MongoDB but it does not reflect the webpage  in production mode
